vsintegration/src/FSharp.Editor/Navigation/GoToDefinitionService.fs

```diff
@@ -284,9 +285,24 @@ type internal FSharpGoToDefinitionService
                         let! implTextSpan = RoslynHelpers.TryFSharpRangeToTextSpan (implSourceText, targetRange)
                         let navItem = FSharpNavigableItem (implDocument, implTextSpan)
                         return navItem
+            // Definition is external
+            | _ -> 
+                let fsSymbol = 
+                    checkFileResults.GetSymbolUseAtLocation(fcsTextLineNumber, lexerSymbol.Ident.idRange.EndColumn, textLine.ToString (), lexerSymbol.FullIsland) 
+                    |> Async.RunSynchronously
+                match fsSymbol with
+                | Some symbol ->
+                    let sigDocument, range = 
+                        match signatureMetadataService.TryFindMetadataRange (originDocument, ast, symbol) with
```

Comment:
`signatureMetadataService.TryFindMetadataRange` should return `asyncMaybe`. It will allow to remove all `Async.RunSynchronously` and 
```fsharp
let signatureText = (sigDocument.GetTextAsync() |> Async.RunTaskSynchronously).ToString()
```
from `tryCreateMetadataContext`